core/container.go

```diff
@@ -1233,6 +1235,19 @@ func (container *Container) Publish(
 		opts[string(exptypes.OptKeyForceCompression)] = strconv.FormatBool(true)
 	}
 
+	clientMetadata, err := engine.ClientMetadataFromContext(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	if clientMetadata.SourceDateEpoch != nil {
+		opts[string(exptypes.OptKeySourceDateEpoch)] = strconv.FormatInt(clientMetadata.SourceDateEpoch.Unix(), 10)
```

Comment:
We definitely need tests for this - they'd belong in `core/integration/container_test.go`/etc (but yes, we should make sure we have a solid idea of the API first!)

The code around anything interacting with buildkit is going through a *lot* of churn right now, so we need to make sure we're not accidentally bricking anything in the future :yellow_heart: